### ISSUE TITLE
Determine how to index external papers/documents not present as DataONE objects

### ISSUE THREAD

**Issue by csjx** [DataONEorg/sem-prov-design#77]
DataONE largely stores science data objects (like spreadsheets, databases, images, etc.) and science metadata objects (usually descriptions of science data in various metadata standards (FGDC, EML, Dublin Core, ISO, etc.).  After data are used in a scholarly publication, the journal article or report likely will not be uploaded to DataONE as a first class object with a DataONE identifier, but will rather likely be identified using an external DOI or other persistent identifier that is not part of the DataONE system.  Our goal is to be able to make provenance statements that show how scholarly papers were derived from particular data objects within DataONE.  One way to make these statements is to bind them to a given data object using specialized properties, like:
<pre>
dataset.1.1 hasProvONEDocument doi:10.xxx/foo222
</pre>
Determine the best way to make provenance statements that relate data objects with DataONE identifiers with provone:Document instances that have identifiers outside of the DataONE system.

**Comment by leinfelder** [comment]
NBD, if the object in question has a URI, then you assert anything you want about it in a prov model, in a semantic annotation, etc. I've long argued that these things need not be hosted in DataONE in order to reference them. Maybe they won't actually exist down the road, but they do/did have a URI and that's better than nothing.

**Comment by PaoloMissier** [comment]
have you looked at the work in the "publishing data services" WG @ RDA?
https://rd-alliance.org/groups/rdawds-publishing-data-services-wg.html


"The WG will create a common, open, universal cross-referencing service for research

articles and research data.

...
7. The cross-referencing service delivered by the WG will be open and free of charge. 

Terms for access to the service will be openly available and non-discriminatory.

**Comment by sycao5** [comment]
For this problem, I may consider the following provenance fields which can be captured by Run Manager:

1. prov:wasAssociatedWith 
            (to link the external paper/document with the related data object including scripts and science data which are present in DataONE)
2. prov: atLocation
            (option1: to store the permanent identifier such as doi for external paper/document)
3. prov: hasPrimarySource
            (option2: to store the permanent identifier such as doi for external paper/document)
4. prov: used
            (to provided a link for other paper/document which cross-reference the current paper/document)

**Comment by sycao5** [comment]
1.CiTO (Citation Typing Ontology)
   URL: http://www.essepuntato.it/lode/http://purl.org/spar/cito
2.Simple Knowledge Organization System (SKOS) 
   URL: http://www.w3.org/TR/2009/REC-skos-reference-20090818/
3.Dublin Core Metadata 
   URL: http://dublincore.org
4.Bibliographic Framework Initiative
   URL for vocabulary:  http://bibframe.org/vocab-list/

**Comment by ludaesch** [comment]
Hi Chris,

Coming a bit late to the party. Just looked at the original issue and trying to understand it.
Provenance models (including PROV) tend to talk about relations such as:
- (Data) --wasGeneratedBy--> [Execution]
- [Execution] --used--> (Data)
- (Data) --wasDerivedFrom--> (Data)
- [Program] --wasDerivedFrom --> [Program]    // oops? OK to use same relation here?

Do I understand the issue correctly: 
- You're saying that any of these four relationships (or exactly these four) may involve a DataONE-internal or an DataONE-external reference!?

Also: can't we simply use the "atLocation" property to point to either an internal (DataONE) or external (URI-, DOI-, or any_other_id_scheme-based) object? I would assume that the different identifier- or reference-schemes being used make it obvious whether an internal or external object is being referenced!? 

What exactly is the problem or what am I missing?

Thanks,

Bertram

**Comment by csjx** [comment]
After discussion in the last meeting, we decided that as long as we have URI pointers to publications (as Ben mentioned above), we can use those URIs as objects of triples, and be sure to type them as `provone:Document` instances:
<pre>
&lt;http://dx.doi.org/10.5061/blah.6gr7t/2&gt;
    a provone:Document;
.
</pre>
Bertram, the difference between 'internal' and 'external' links for DataONE is that objects that are registered in the Coordinating Node (ie uploaded to a Member Node and then harvested) will be available via the [CNRead.resolve()](http://jenkins-1.dataone.org/jenkins/job/API%20Documentation%20-%20trunk/ws/api-documentation/build/html/apis/CN_APIs.html#CNRead.resolve) API call.  when we construct resource maps, DataONE has a [policy](http://jenkins-1.dataone.org/jenkins/job/API%20Documentation%20-%20trunk/ws/api-documentation/build/html/design/DataPackage.html#generating-resource-maps) of only allowing CN resolve URIs as triple subjects for objects.  'External' (ie non-CN-resolvable) URIs might make the retrieval of object bytes brittle.  However, it sounds like in the case of publications, DOI-type URIs should be fine.  during implementation of this in software, we'll have to determine if the policy needs to change. Closing this issue.
